### PR TITLE
#59 #60 #61 integrate front-end with back-end and  CandidatePage 

### PR DESCRIPTION
**Pull request by Tasneem-Gad**

### CODE REVIEW

File: ui/src/app/Pages/candidates/candidates.component.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  selector: 'app-candidates',
+  templateUrl: './candidates.component.html',
+  styleUrls: ['./candidates.component.sass']
+})
+export class CandidatesComponent implements OnInit {
+
+  constructor(private fb: FormBuilder) { }
+  candidateForm : FormGroup;
+  ngOnInit(): void {
+    this.candidateForm = this.fb.group({
```
Comment (ZinabHashem) [inline ui/src/app/Pages/candidates/candidates.component.ts:14]: make form in outside function and call it in ngOnInit 
to use single principle responsibility

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

File: ui/src/app/Pages/candidates/candidates.component.ts (hunk 1 above)
Comment (Tasneem-Gad) [reply ui/src/app/Pages/candidates/candidates.component.ts:17]: okay

File: ui/src/app/Pages/AddCandidateInfo/candidates/candidates.component.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      picture: [''],
+    });
+  }
+  saveCandidate() {
+    this.candidateDetails.id = this.candidateForm.value.id;
+    this.candidateDetails.name = this.candidateForm.value.name;
+    this.candidateDetails.phone = this.candidateForm.value.phone;
+    this.candidateDetails.email = this.candidateForm.value.email;
+    this.candidateDetails.documentId = this.candidateForm.value.documentId;
+    this.candidateDetails.picture = this.candidateForm.value.picture;
```
Comment (ZinabHashem) [inline ui/src/app/Pages/AddCandidateInfo/candidates/candidates.component.ts:66]: you don't need to make this lines
you can send form.value direct

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

File: ui/src/index.html [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
-  <meta charset="utf-8">
-  <title>CandiMeter</title>
-  <base href="/">
-  <meta name="viewport" content="width=device-width, initial-scale=1">
-  <link rel="icon" type="image/x-icon" href="favicon.ico">
+    <meta charset="utf-8">
+    <title>CandiMeter</title>
+    <base href="/">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <!-- <link rel="icon" type="image/x-icon" href="favicon.ico"> -->
```
[the remark this answers is not in the document]
Comment (Tasneem-Gad) [reply ui/src/index.html:9]: we don't need  this in our projects

File: ui/src/app/core/core.module.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ import { RouterModule } from '@angular/router';
 import { MainComponent } from './components/main/main.component';
 import { SideBarListComponent } from './components/side-bar-list/side-bar-list.component';
 import { HomePageComponent } from './components/home-page/home-page.component';
+import { HTTP_INTERCEPTORS } from '@angular/common/http';
```
[the remark this answers is not in the document]
Comment (Tasneem-Gad) [reply ui/src/app/core/core.module.ts]: okay

File: ui/src/app/domain/candidate/candidate.repository.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  private readonly url = `${SettingsService.configurationEnvironment.api.baseUrl}`;
+  constructor(protected  httpClient: HttpClient) {
+    super(httpClient);
+  }
+
+
+   getResourceUrl(): string{
+    return 'candidate';
+   };
+   getListByCandidate(p: {} = {}): Observable<Response<Candidate>> {
```
[the remark this answers is not in the document]
Comment (Tasneem-Gad) [reply ui/src/app/domain/candidate/candidate.repository.ts]: okay

File: ui/src/app/Pages/AddCandidateInfo/candidates/candidates.component.ts (hunk 2 above)
Comment (Tasneem-Gad) [reply ui/src/app/Pages/AddCandidateInfo/candidates/candidates.component.ts]: I will do that.

File: ui/src/app/Pages/pages.module.ts [hunk 6]
```diff
@@ -26,12 +29,15 @@ const routes: Routes = [
...
+    CandidatesComponent
   ],
   imports: [
     RouterModule.forChild(routes),
     CommonModule,
+    ReactiveFormsModule,
     SharedModule,
-    ReactiveFormsModule
-    ]
+    RouterModule.forChild(routes)
```
[the remark this answers is not in the document]
Comment (Tasneem-Gad) [reply ui/src/app/Pages/pages.module.ts]: I will remove this .

File: ui/src/index.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (ZinabHashem) [reply ui/src/index.html]: if you don't need delete this not leave comment

File: ui/src/app/Pages/AddCandidateInfo/candidates/candidates.component.html [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+              <div><img [src]="dataimage" width="300px" /></div>
+            </div>
+          </div>
+          <div fxLayoutAlign="center center">
+            <button type="submit" mat-raised-button type="submit" color="primary" fxFlex.lt-lg="80%" fxFlex="30"
+              [disabled]="submit">
+              {{candidateForm.controls['id'].value ? 'Update Candidate':'Add Candidate'}}
+            </button>
+
+            <div *ngIf="isAppear; then appear"></div>
```
[the remark this answers is not in the document]
Comment (Tasneem-Gad) [reply ui/src/app/Pages/AddCandidateInfo/candidates/candidates.component.html]: okay

**Pull request merged**